core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -442,7 +452,7 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
       numPendingExecutors += knownExecutors.size
     }
 
-    doKillExecutors(executorsToKill)
+    (force || !executorsToKill.isEmpty) && doKillExecutors(executorsToKill)
```

Comment:
It will still work; we'll just send an empty request to the master, not a big deal. We can also check whether it's empty here but we'll need to update the docs.